layouts/Default.vue

```diff
@@ -127,6 +131,20 @@ export default {
     this.$bus.$on('offline-order-confirmation', this.onOrderConfirmation)
 
     this.$store.commit('ui/setSupportsWebp', supportWebp())
+
+    const show = (message = null) => {
+      this.loaderMessage = message
+      this.$store.commit('ui/setLoader', true)
+    }
+    const hide = () => {
+      this.$store.commit('ui/setLoader', false)
+    }
+    this.$bus.$on('notification-progress-start', show)
+    this.$bus.$on('notification-progress-stop', hide)
+    this.$on('hook:beforeDestroy', () => {
+      this.$bus.$off('notification-progress-start', show)
+      this.$bus.$off('notification-progress-stop', hide)
+    })
```

Comment:
it's not best place to handle loader :thinking: 
Suggestion:
```
import { store } from '@vue-storefront/core/app'
import EventBus from '@vue-storefront/core/compatibility/plugins/event-bus'
    const show = (message = null) => {
      store.commit('ui/setLoaderMessage', message) // new commit
      store.commit('ui/setLoader', true)
    }
    const hide = () => {
      store.commit('ui/setLoader', false)
    }
    const mounted = () => {
    EventBus.$on('notification-progress-start', show)
    EventBus.$on('notification-progress-stop', hide)
};
const beforeDestroy = () => {
      EventBus.$off('notification-progress-start', show)
      EventBus.$off('notification-progress-stop', hide)
};
```
and add it to loader component (you can add mounted and beforeDestory hooks to functonal component
https://dev.to/denisinvader/mounted-and-beforedestroy-hooks-in-vuejs-functional-components-7bi)

Next pass to loader component message by using getter
```
<loader :message="getLoaderMessage"
```